src/PosthogAnalytics.ts

```diff
@@ -285,6 +354,11 @@ export class PosthogAnalytics {
       : Anonymity.Disabled;
     this.setAnonymity(anonymity);
     if (anonymity === Anonymity.Pseudonymous) {
+      this.setRegistrationType(
+        window.matrixclient.isGuest() || window.isPasswordlessUser
```

Comment:
Can i get acces to `isPasswordlessUser` from the client peg? It seems like this is a element call specific property not part of the client interface?

Reply:
Ah, no you can't since the js-sdk doesn't know about it. You would indeed need the context, or which is impossible using a shared instance method - we have an impedance mismatch between singletons and contexts. We should probably decide on one or the other rather than mixing them, but this is probably an okay workaround for now.